plugins/csp-trajectories/src/DeepSpaceDot.hpp

```diff
@@ -20,13 +19,47 @@
 namespace csp::trajectories {
 
 /// A deep space dot is a simple marker indicating the position of an object, when it is too
-/// small to see.
+/// small to be seen. It can be used as a HUD element (a colorful circle which is drawn after tone
+/// mapping) or as a flare (a bright glowing circle which is drawn before tone mapping).
 class DeepSpaceDot : public IVistaOpenGLDraw {
  public:
-  cs::utils::Property<VistaColor> pColor = VistaColor(1, 1, 1); ///< The color of the marker.
+  /// The mode of the marker.
+  enum class Mode {
+    /// In this mode, the dot is drawn as an antialiased circle with the given color.
+    eSmoothCircle,
+
+    /// In this mode, the dot is drawn with a linear brightness gradient with an average luminance
+    /// of the given value.
+    eHDRBillboard,
+
+    /// In this mode, the dot is drawn ten times larger than the given solid angle and an
+    /// exponential glow effect is applied to the dot. This is useful to add an artificial glow to
+    /// an object in non-HDR mode.
+    eFlare,
+  };
+
+  /// Shows or hides the marker.
+  cs::utils::Property<bool> pVisible = true;
+
+  /// The mode of the marker.
+  cs::utils::Property<Mode> pMode = Mode::eSmoothCircle;
 
-  DeepSpaceDot(std::shared_ptr<Plugin::Settings> pluginSettings,
-      std::shared_ptr<cs::core::SolarSystem>     solarSystem);
+  /// When to draw the dot.
+  cs::utils::Property<int> pDrawOrder =
+      static_cast<int>(cs::utils::DrawOrder::eTransparentItems) - 1;
+
+  /// The color of the marker. In eHDRFlare mode, this corresponds to the average albedo of the
+  /// object.
+  cs::utils::Property<VistaColor> pColor = VistaColor(1, 1, 1, 1);
```

Comment:
Why are we using `VistaColor` here instead of a `glm::vec4`? We should use glm everywhere possible for consistency. It also would be easier to pass to the shader.